comp/metadata/internal/util/inventory_payload.go

```diff
@@ -102,14 +102,18 @@ type InventoryPayload struct {
 
 // CreateInventoryPayload returns an initialized InventoryPayload. 'getPayload' will be called each time a new payload
 // needs to be generated.
-func CreateInventoryPayload(conf config.Component, l log.Component, s serializer.MetricSerializer, getPayload PayloadGetter, flareFileName string) InventoryPayload {
+func CreateInventoryPayload(conf config.Component, l log.Component, s serializer.MetricSerializer, getPayload PayloadGetter, flareFileName string, customInterval time.Duration) InventoryPayload {
 	minInterval := time.Duration(conf.GetInt("inventories_min_interval")) * time.Second
-	if minInterval <= 0 {
+	if customInterval > 0 {
+		minInterval = customInterval
+	} else if minInterval <= 0 {
```

Comment:
What about adding to public method `SetMinInterval` and `SetMaxInterval` ? This way the `CreateInventoryPayload` keeps its argument list for mandatory parameters and payload can set a custom interval for min or max.